### PR TITLE
#2219 - Activity Log Table Maintenance - Data Cleanup Button

### PR DESCRIPTION
**Pull request by kodinkat**
- fixes: #2219 
    - A "Data cleanup" button that runs an sql query to delete activity from fields that have been deleted.

Screenshot 2023-10-16 at 12 44 03

### CODE REVIEW

**Comment by corsacca** [comment]
@kodinkat Let's have the first table be "generation counts" instead of "field"
And add a second table for "activity data"

**Comment by corsacca** [comment]
I like the progress: done column :)

File: dt-core/admin/menu/tabs/admin-endpoints.php [hunk 1]
```diff
@@ -147,6 +156,57 @@ public function update_custom_field_translations( WP_REST_Request $request ){
...
+            // Build list of existing field ids.
+            $field_settings = DT_Posts::get_post_field_settings( $post_type, false );
+            $field_ids = array_keys( $field_settings );
+
+            // Identify deleted field activity logs.
+            global $wpdb;
+
+            $id_sql = [];
+            foreach ( $field_ids as $id ) {
+                $id_sql[] = "(log.object_subtype NOT LIKE '" . $id . "%')";
```
Comment (corsacca) [inline dt-core/admin/menu/tabs/admin-endpoints.php:173]: @kodinkat what is the `%` wildcard at the end here for?

Comment (kodinkat) [reply dt-core/admin/menu/tabs/admin-endpoints.php:173]: @corsacca to accommodate `communication_channel` field type naming conventions.

Screenshot 2023-10-20 at 11 00 25

**Comment by corsacca** [comment]
Thanks @kodinkat!

**Pull request merged**